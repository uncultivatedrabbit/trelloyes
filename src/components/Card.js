import React from "react";
import '../css/Card.css'

export default function Card(props) {
  return (
    <div className="Card">
      <button onClick={() => props.handleDeleteCard(props.id)}type="button">delete</button>
      <h3>{props.title}</h3>
      <p>{props.content}</p>
    </div>
  );
}
