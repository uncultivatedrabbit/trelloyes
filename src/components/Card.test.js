import React from "react";
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import Card from "./Card";

describe('Card component', () => {
	
	// Smoke test
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<Card title="title" content="content" />, div);
		ReactDOM.unmountComponentAtNode(div);
	});
	
	// Snapshot
	it('renders a card as expected', () => {
		const tree = renderer.create(
			<Card title="title" content="content" />
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
	
});