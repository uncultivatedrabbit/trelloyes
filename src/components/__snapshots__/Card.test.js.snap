// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders a card as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    title
  </h3>
  <p>
    content
  </p>
</div>
`;

exports[`renders the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3 />
  <p />
</div>
`;
